static/app/components/events/autofix/autofixDiff.tsx

```diff
@@ -91,32 +98,377 @@ function HunkHeader({lines, sectionHeader}: {lines: DiffLine[]; sectionHeader: s
+      return line.source_line_no;
+    }
+    if (lineType === DiffLineType.ADDED) {
+      // Find the first non-null target_line_no
+      for (let i = index; i < linesWithChanges.length; i++) {
+        if (linesWithChanges[i].target_line_no !== null) {
+          return linesWithChanges[i].target_line_no;
+        }
+      }
+    }
+    return null;
+  };
+
+  const handleClearChanges = () => {
+    setEditedContent('');
+    setEditedLines([]);
+  };
+
   return (
     <Fragment>
       <HunkHeaderEmptySpace />
       <HunkHeader lines={lines} sectionHeader={header} />
-      {linesWithChanges.map(line => (
-        <Fragment key={line.diff_line_no}>
+      {linesWithChanges.map((line, index) => (
+        <Fragment key={index}>
           <LineNumber lineType={line.line_type}>{line.source_line_no}</LineNumber>
           <LineNumber lineType={line.line_type}>{line.target_line_no}</LineNumber>
           <DiffContent
             lineType={line.line_type}
             data-test-id={makeTestIdFromLineType(line.line_type)}
           >
             <DiffLineCode line={line} />
+            {lineGroups.some(group => index === group.start) && (
+              <ButtonGroup>
+                <ActionButton
+                  size="xs"
+                  icon={<IconEdit size="xs" />}
+                  aria-label={t('Edit changes')}
+                  title={t('Edit')}
+                  onClick={() => handleEditClick(index)}
+                />
+                <ActionButton
+                  size="xs"
+                  icon={<IconClose size="xs" />}
+                  aria-label={t('Reject changes')}
+                  title={t('Reject')}
+                  onClick={() => rejectChanges(index)}
+                />
+              </ButtonGroup>
+            )}
+            {editingGroup === index && (
+              <EditOverlay ref={overlayRef}>
+                <OverlayTitle>{t('Editing %s', fileName)}</OverlayTitle>
+                <SectionTitle>
+                  {t(
+                    '%s deleted line%s%s',
+                    linesWithChanges
+                      .slice(index, lineGroups.find(g => g.start === index)?.end! + 1)
+                      .filter(l => l.line_type === DiffLineType.REMOVED).length,
+                    linesWithChanges
+                      .slice(index, lineGroups.find(g => g.start === index)?.end)
+                      .filter(l => l.line_type === DiffLineType.REMOVED).length === 1
+                      ? ''
+                      : 's',
+                    linesWithChanges
+                      .slice(index, lineGroups.find(g => g.start === index)?.end)
+                      .filter(l => l.line_type === DiffLineType.REMOVED).length > 0
+                      ? t(
+                          ' from line %s',
+                          getStartLineNumber(index, DiffLineType.REMOVED)
+                        )
+                      : ''
+                  )}
```

Comment:
kind of a handful to read, maybe cleaner to make these variables before the jsx code and then use it here, something like `const deletedLineTitle = t(...`?